src/common/low_precision_transformations/src/recurrent_cell.cpp

```diff
@@ -0,0 +1,215 @@
+        {X_in, H_in, C, S, W_in, R_in, B});
+    const auto gru_seq  = ngraph::pattern::wrap_type<ngraph::opset5::GRUSequence>(
+        {X_in, H_in,    S, W_in, R_in, B});
+
+    ngraph::graph_rewrite_callback callback = [this](pattern::Matcher& m) {
+        auto op = m.get_match_root();
+        if (transformation_callback(op)) {
+            return false;
+        }
+
+        return transform(*context, m);
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(
+        std::make_shared<pattern::op::Or>(
+            OutputVector {
+                lstm_seq,
+                gru_seq
+            }),
+        "RecurrentCellTransformation");
+
+    this->register_matcher(m, callback);
+}
+
+bool RecurrentCellTransformation::transform(TransformationContext& context, ngraph::pattern::Matcher& m) {
+    const auto lstm = m.get_match_root();
+    if (!canBeTransformed(context, lstm)) {
+        return false;
+    }
+    for (size_t parentIndex = 0ul; parentIndex < lstm->get_input_size(); parentIndex++) {
+        auto lstm_parent = lstm->get_input_node_shared_ptr(parentIndex);
+        if (is_type<ngraph::opset1::FakeQuantize>(lstm_parent)) {
+            auto fq_parent = lstm_parent->get_input_node_shared_ptr(0);
+            if (is_type<ngraph::opset5::Constant>(fq_parent)) {
+                auto fq_node = as_type_ptr<ngraph::opset1::FakeQuantize>(lstm_parent);
+                const QuantizationDetails quantizationDetails = QuantizationDetails::getDetails(fq_node);
+                const auto precisionsAttribute = getAttributeFromOutput<PrecisionsAttribute>(lstm_parent);
+                const auto precisions = precisionsAttribute.empty()
+                                            ? defaultPrecisions
+                                            : precisionsAttribute.as<PrecisionsAttribute>().value();
+                const DataPrecision dataPrecision = getDataPrecision(lstm_parent, quantizationDetails, precisions);
+                auto QDQ = NetworkHelper::decomposeFakeQuantize(fq_node,
+                                                                  dataPrecision.precision,
+                                                                  dataPrecision.min,
+                                                                  dataPrecision.max,
+                                                                  dataPrecision.hasZeroPoint,
+                                                                  updatePrecisions);
+                std::shared_ptr<ngraph::Node> new_fq = std::get<0>(QDQ);
+                std::shared_ptr<ngraph::Node> deq_multiply = std::get<1>(QDQ);
+                if (deq_multiply == nullptr || new_fq == nullptr) {
+                    return false;
+                }
+
+                auto multiply_parent = deq_multiply->get_input_node_shared_ptr(0);
+                if (is_type<ngraph::opset1::Subtract>(multiply_parent)) {
+                    return false;
+                }
```

Comment:
As you mentioned before, dequantization shifts for data input are supported. So we shouldn't return false if `parentIndex` is 0

Reply:
Done. Please double check